gateway-service-definitions/src/main/resources/services/yarnuiv2/3.0.0/rewrite.xml

```diff
@@ -170,5 +170,13 @@
       <apply path="Location" rule="YARNUIV2/yarnuiv2/outbound/headers/index/location"/>
     </content>
   </filter>
+   
+  <rule dir="IN" name="YARNUIV2/yarnuiv2/inbound/proxy" pattern="*://*:*/**/yarnuiv2/proxy/{**}?{**}">
+      <rewrite template="{$serviceUrl[YARNUIV2]}/proxy/{**}?{**}"/>
+   </rule>
+
+   <rule dir="OUT" name="YARNUIV2/yarnuiv2/outbound/proxy" pattern="*://*:*/proxy/{**}">
+      <rewrite template="{gateway.scheme}://{gateway.host}:{gateway.port}/gateway/default/sparkhistory/history/{**}?{**}"/>
```

Comment:
This isn't correct for a few reasons.
* This assumes the default topology due to `gateway/default`
* This will cause all proxying to go to Spark history server and will break mapreduce history server links.

Reply:
Right, also the rule "YARNUIV2/yarnuiv2/outbound/proxy" does not seem to be references anywhere, how is this getting triggered ? you might also want to try a more specific pattern for this rule, currently it is  `*://*:*/proxy/{**}`

Reply:
Ok, Let me check on both and get back.